app/workers/deletes.php

```diff
@@ -92,6 +95,27 @@ public function shutdown(): void
     {
     }
 
+    /**
+     * @param Document $document teams document
+     * @param string $projectId
+     */
+    protected function deleteCollection(Document $document, string $projectId): void
+    {
+        $collectionId = $document->getId();
+
+        $dbForInternal = $this->getInternalDB($projectId);
+
+        $this->deleteByGroup('attributes', [
+            new Query('collectionId', Query::TYPE_EQUAL, [$collectionId])
+        ], $dbForInternal);
+
+        $this->deleteByGroup('indexes', [
+            new Query('collectionId', Query::TYPE_EQUAL, [$collectionId])
+        ], $dbForInternal);
+
+        $dbForInternal->deleteDocument('collections', $collectionId);
```

Comment:
I was pretty sure this delete was the only one that should have actually be called in the HTTP call, and the drop table call was the one to be called here. Did I miss anything?

Reply:
Good catch - bfdf6de was the incorrect solution and was technically duplicated code (`deleteDocument` is indeed called in the deleteCollection route). Updated in cb7c810